### DISCUSSION TITLE
PorousFlow Basic question on BCs

### DISCUSSION THREAD

**Discussion by burritotukituki** [idaholab/moose#25468] [Q&A Modules: General]
Hi,

In my TH model, I fix the temperature at the boundaries and set a zero-pressure condition at the top of the model. I would expect a pressure profile given by -rho * g * y, but it looks like the solution found in steady state is slightly different. Why is that? Perhaps there is some physics I am missing? (see temp_grav_test_s.i)

Then, I modify the BCs to have flux from left to right with Dirichlet BCs. The temperature is fixed at the left, and 'PorousFlowOutflowBC' at the right allows free outflow of temperature. It seems to work fine for pore pressure, but the temperature diverges. Note that this works fine without gravity. Why is that? (see temp_grav_test.i)

Finally, I would like to implement 'PorousFlowPiecewiseLinearSink' to model flow and temperature instead of Dirichlet BCs. How can I set up these boundaries in the case of temperature? Something like this, maybe?
```
  [in_left]
    type = PorousFlowPiecewiseLinearSink
    variable = porepressure
    boundary = 'left'
    pt_vals = '-1e9 1e9' # x coordinates defining g
    multipliers = '-1e9 1e9' # y coordinates defining g
    PT_shift = 2.E6   # BC pressure
    flux_function = 1E-5 # Variable C
    fluid_phase = 0
  []

  [in_left_temp]
    type = PorousFlowPiecewiseLinearSink
    variable = temperature
    boundary = 'left'
    pt_vals = '-1e9 1e9' # x coordinates defining g
    multipliers = '-1e9 1e9' # y coordinates defining g
    PT_shift = 280   # BC temperature
    flux_function = 1E-5 # Variable C
    fluid_phase = 0
  []
```
To the models: https://drive.google.com/drive/folders/1KY9KZ558KDBJsSsjcUQEHvs1R0v5FTed?usp=drive_link
Edit:
Link to the github repos: https://github.com/burritotukituki/PorousFlowStuff

Many thanks in advance!
Tukituki

**Comment by GiudGiud** [comment]
Hello

The google drive seems to have restricted access. We'd prefer a github repo in general to host input files. 

Did you specify a gravity term?

Guillaume

**Reply by burritotukituki** [reply]
Hi,
Thanks I added a github repo. Yes, I specify gravity ;).

**Reply by GiudGiud** [reply]
I cant see it. Not sure if it s morning-me though haha

**Reply by burritotukituki** [reply]
Here
https://github.com/burritotukituki/PorousFlowStuff/blob/main/temp_grav_test.i
and here
https://github.com/burritotukituki/PorousFlowStuff/blob/main/temp_grav_test_s.i

**Reply by GiudGiud** [reply]
I dont see what could be wrong. @cpgr mind taking a look

**Comment by 1runer** [comment]
I wonder, if you considered that SimpleFluidProperties calculates fluid density as a function of pressure and temperature? The given density0 is just the reference density and gets updated during solution.
So your result of '-rho * g * y' has to assume the correct density as function of y.

**Reply by cpgr** [reply]
This is the most likely cause - you can set the thermal expansion to 0 and make the bulk modulus really big to try and mimic a constant density

**Reply by burritotukituki** [reply]
That makes a lot of sense! Thanks :)

Any comments on the BCs? 
@cpgr 
@1runer

**Reply by 1runer** [reply]
I'm avoiding telling you something wrong/not fully true. 
The BCs implemented in PorousFlow are very complex and have many parameters to play with.

@cpgr or @WilkAndy are the experts on this and can certainly help.

**Comment by burritotukituki** [comment]
Looks like @WilkAndy is an expert on this field. Can you please give me a hint on this?
Cheers!

**Reply by burritotukituki** [reply]
Hi @cpgr, can you please also help in in this issue?
